modules/debts_api/app/controllers/debts_api/v0/digital_disputes_controller.rb

```diff
@@ -24,12 +27,31 @@ def create
       private
 
       def process_submission
-        service = DigitalDisputeSubmissionService.new(current_user, submission_params[:files])
+        metadata = parse_metadata(submission_params[:metadata])
+
+        service = DebtsApi::V0::DigitalDisputeSubmissionService.new(
+          current_user,
+          submission_params[:files],
+          metadata
+        )
         service.call
       end
 
+      def parse_metadata(metadata_param)
+        return nil if metadata_param.blank?
+        return metadata_param if metadata_param.is_a?(Hash)
+
+        JSON.parse(metadata_param, symbolize_names: true)
+      rescue JSON::ParserError => e
+        Rails.logger.error "[DIGITAL_DISPUTES_CONTROLLER ERROR] Failed to parse metadata JSON: #{e.message}"
+        nil
+      end
+
       def submission_params
-        params.permit(files: [])
+        params.permit(
+          :metadata,
```

Comment:
Looking at the `:metadata` param and then the parse_metatadata method.

I wonder if this would clean things up:
```
params.permit(
          metadata: {},
          files: []
          )
```

With that, maybe we could get rid of the JSON parsing in parse_metadata.

Also, we return `nil` if metadata_param is blank. Then we do truthy/falsey checks later on. I wonder if we made that change to the params and an empty hash was passed along if we could get rid of the checks for truthy-ness.